src/contracts/campaigns/BraidCampaign.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity ^0.4.4;
+
+
+import "utils/Owned.sol";
+import "verifiers/OwnedVerifier.sol";
+import "tokens/StandardToken.sol";
+
+/*
+A custom WeiFund compatable campaign for the Braid the Movie Ethereum campaign.
+*/
+
+contract BraidCampaign is Owned, StandardToken {
+  modifier pastFreezeDate {
+    // now must be greater than freeze date
+    if (block.number >= freezePeriod) {
+      _;
+    } else {
+      throw;
+    }
+  }
+
+  function transfer(address _to, uint256 _value) pastFreezeDate public returns (bool) {
+    return super.transfer(_to, _value);
+  }
+
+  function transferFrom(address _from, address _to, uint256 _value) pastFreezeDate public returns (bool success) {
+    return super.transferFrom(_from, _to, _value);
+  }
+
+  function approve(address _spender, uint256 _value) pastFreezeDate public returns (bool success) {
+    return super.approve(_spender, _value);
+  }
+
+  function () payable public {
+    // allow fallback function.. contribute as normal
+    contributeMsgValue(defaultAmounts);
+  }
+
+  function contributeMsgValue(uint256[] _amounts) payable public returns (uint256) {
+    // the total tokens amount to be issued to the sender
+    uint256 tokensOwed = msg.value / price;
```

Comment:
Dino wants to grant token to have more precision than this. I just did this as this is our current price calculation. I'm talking with him now. He says he wants tokens that are divisible and have more "decimals" than this. So I'm assuming we allow more precision from the msg.value number.

I'm not sure how to do this math, and I can't believe this was not discussed before. A huge oversight. So we just need to work up a scheme with more precision, and make sure the contributor gets more of their contributions worth of tokens given the USD price.